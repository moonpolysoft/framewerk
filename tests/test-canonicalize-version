#! /bin/sh

(
cd ..
set -e

# If it (epoch) is omitted then the upstream_version may not contain any colons.
test `fw-exec package/deb/canonicalize-version 1flass:turg` = 1flass.turg
test `fw-exec package/deb/canonicalize-version 0:1flass:turg` = 0:1flass:turg

# if it (debian revision) isn't present then the upstream_version may not contain a hyphen. 
test `fw-exec package/deb/canonicalize-version 1flass:turg-*xx` = 1flass.turg..xx
test `fw-exec package/deb/canonicalize-version 1flass:turg-xxx` = 1flass.turg-xxx
test `fw-exec package/deb/canonicalize-version 0:1flass:turg-*xx` = 0:1flass:turg..xx
test `fw-exec package/deb/canonicalize-version 0:1flass:turg-xxx` = 0:1flass:turg-xxx

# The upstream_version may contain only alphanumerics[32] and the characters . + - : ~ (full stop, plus, hyphen, colon, tilde) ...

test `fw-exec package/deb/canonicalize-version 1flass*` = 1flass. 

set +e
echo " ****** following errors are expected ****** "
# The upstream_version ... should start with a digit.
fw-exec package/deb/canonicalize-version flass:turg && exit 1
set -e
) > test-canonicalize-version.out 2>&1
