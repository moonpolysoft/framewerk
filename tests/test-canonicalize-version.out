package/deb/canonicalize: warning: adjusting nonconforming upstream_version '1flass:turg' of version '1flass:turg' to '1flass.turg'
package/deb/canonicalize: warning: adjusting nonconforming upstream_version '1flass:turg-*xx' of version '1flass:turg-*xx' to '1flass.turg..xx'
package/deb/canonicalize: warning: adjusting nonconforming upstream_version '1flass:turg' of version '1flass:turg-xxx' to '1flass.turg'
package/deb/canonicalize: warning: adjusting nonconforming upstream_version '1flass:turg-*xx' of version '0:1flass:turg-*xx' to '1flass:turg..xx'
package/deb/canonicalize: warning: adjusting nonconforming upstream_version '1flass*' of version '1flass*' to '1flass.'
 ****** following errors are expected ****** 
package/deb/canonicalize: warning: adjusting nonconforming upstream_version 'flass:turg' of version 'flass:turg' to 'flass.turg'
package/deb/canonicalize-version: fatal: upstream_version component 'flass.turg' of 'flass:turg' needs to start with a digit.
